packages/cli/src/commands/devHandler.js

```diff
@@ -49,8 +51,78 @@ export const handler = async ({
   }
 
   if (side.includes('web')) {
+    let proposedPort = parseInt(getConfig().web.port)
+    const forwardedPortMatches = forward.match(
+      /--port=[0-9][0-9]?[0-9]?[0-9]?[0-9]? ?/
```

Comment:
Not totally sure about all the `?`s, but you can replace [0-9] with just `\d` and a quantifier. We should check for `=` or a space after port:
```suggestion
      /--port(=| )\d{2,4}/
```

Reply:
I've got a feeling the regex for a port is way more complex than that though. I'm not sure if we need to validate the port. Webpack probably does that. We just want to see if they set it, and if they did use that. So maybe we can just check for the `port` flag?

Reply:
So regex isn't an area of strength for me, to be honest I've just never dedicated the time to learn it properly. This particular regex was simply one I found to work but I knew it would almost certainly be possible to improve it.

I felt the port value needed to be extracted because we need to check if that particular port the user requests is available? Knowing they wish to use a custom port isn't sufficient information to check if their port is available. Do you agree?

Reply:
I've come up with a different suggestion for this: https://github.com/redwoodjs/redwood/pull/6437#discussion_r1014529819